application/modules/guest/controllers/gateways/Stripe.php

```diff
@@ -73,53 +73,102 @@ public function create_checkout_session($invoice_url_key)
+        if ($session->payment_status === 'paid') {
+
+            // ----------------------
+            // SUCCESS FLOW
+            // ----------------------
+            // a) Show success message
+            $this->session->set_flashdata(
+                'alert_success',
+                sprintf(trans('online_payment_payment_successful'), $invoice->invoice_number)
+            );
             $this->session->keep_flashdata('alert_success');
 
-            //record the payment
+            // b) Record the payment in the invoiceplane DB
             $this->load->model('payments/mdl_payments');
-
             $this->mdl_payments->save(null, [
-                'invoice_id' => $invoice_id,
-                'payment_date' => date('Y-m-d'),
-                'payment_amount' => $session->amount_total / 100,
+                'invoice_id'        => $invoice_id,
+                'payment_date'      => date('Y-m-d'),
+                'payment_amount'    => $session->amount_total / 100,
                 'payment_method_id' => get_setting('gateway_stripe_payment_method'),
-                'payment_note' => 'payment intent ID: ' . $session->payment_intent,
+                'payment_note'      => 'payment intent ID: ' . $session->payment_intent,
             ]);
 
-            //record the online payment
+            // c) Record the “merchant response”
             $this->db->insert('ip_merchant_responses', [
-                'invoice_id' => $invoice_id,
+                'invoice_id'                   => $invoice_id,
                 'merchant_response_successful' => true,
-                'merchant_response_date' => date('Y-m-d'),
-                'merchant_response_driver' => 'stripe',
-                'merchant_response' => '',
-                'merchant_response_reference' => 'payment intent ID: ' . $session->payment_intent,
+                'merchant_response_date'       => date('Y-m-d'),
+                'merchant_response_driver'     => 'stripe',
+                'merchant_response'            => '',
+                'merchant_response_reference'  => 'payment intent ID: ' . $session->payment_intent,
             ]);
 
+            // d) Redirect back to invoice view
             redirect(site_url('guest/view/invoice/' . $invoice->invoice_url_key));
-        } catch (Error|Exception|ErrorException $e) {
-            //TODO: log error
 
+        } else {
+
+            // ----------------------
+            // FAILURE / CANCELLED FLOW
+            // ----------------------
+            // a) Show an error or “payment not completed” notice to the user
+            //    (Adjust text as you wish)
             $this->session->set_flashdata('alert_error',
-                trans('online_payment_payment_failed') . '<br/>' . $$e->getMessage());
+                sprintf(trans('online_payment_payment_failed')) .
+                '<br/>' .
+                'Payment was not completed. Stripe session status: ' . $session->payment_status
```

Comment:
use as much translations as you can, if a translation is missing, just add it in `/application/language/ip_lang.php`